gsys/src/lib.rs

```diff
@@ -329,8 +329,33 @@ impl TwoHashesWithValue {
     }
 }
 
+/// Current version of execution settings.
+///
+/// Backend maintains backward compatibility with previous versions of execution
+/// settings. This structure matches to the most recent version of execution
+/// settings supported by backend.
+#[repr(C, packed)]
+#[derive(Debug, Clone, Copy)]
+pub struct ExecSettings {
```

Comment:
I would make name like this to clearly indicate version
```suggestion
pub struct ExecSettingsV1 {
```